### ISSUE TITLE
Metodę doLogin oraz doLoginEnc oznaczyliśmy jako deprecated - korzystaj z OAuth2

### ISSUE THREAD

**Issue by PawelTaberski** [allegro/allegro-api#911]
Metodę [doLogin](https://allegro.pl/webapi/documentation.php/show/id,82) oraz [doLoginEnc](https://allegro.pl/webapi/documentation.php/show/id,83) oznaczyliśmy dziś jako **deprecated. Nie zalecamy korzystania z nich w nowych integracjach, a w już istniejących rozwiązaniach mocno zachęcamy do migracji na [OAuth2](https://developer.allegro.pl/auth/).** 
W WebAPI, aby otrzymać identyfikator sesji korzystaj z [doLoginWithAccessToken](https://allegro.pl/webapi/documentation.php/show/id,101582).

OAuth2 jest dla Allegro API (WebAPI i REST API) podstawową metodą autoryzacji aplikacji i użytkowników. 
W ramach rozbudowy możliwości Allegro OAuth [przygotowaliśmy nową ścieżkę](https://developer.allegro.pl/auth/#DeviceFlow), dzięki której umożliwiliśmy autoryzację użytkownika na urządzeniach lub w aplikacjach, które nie posiadają:

- interfejsu graficznego,
- przeglądarki,
- lub innego prostego sposobu na wpisanie tekstu.

Dzięki temu każda aplikacja powinna być w stanie skorzystać z OAuth.
Jeśli obecnie korzystasz z metody **doLogin** lub **doLoginEnc**, zaplanuj migrację na nowsze rozwiązanie. Docelowo będziemy wygaszali te metody. O planowanej dacie ich wygaszenia poinformujemy w osobnym komunikacie.

**Comment by marcinja2** [comment]
Zastanawiam się czy bez konkretnej domeny i stałego adresu ip za bramą NAT na sieci prywatnej i każdym jej komputerze w związku ze zmianą będzie możliwe logowanie? jestem początkujący i jeszcze się uczę nowych rozwiązań wiec jeśli pytanie wydaje się głupie to i tak proszę o odp

**Comment by PawelTaberski** [comment]
@marcinja2 Tak, będzie to możliwe, na komputerze w sieci wewnętrznej - np. jako redirect_uri przekaż adres z sieci wewnętrznej (np. http://firmowy-serwer1.local),  localhost albo adres IP.
Wtedy przeglądarka skieruje użytkownika na taki adres, ważne, żeby użytkownik, który się loguje miał tam dostęp.
Przemyśl, czy w takim wypadku nie skorzystać z nowego flow autoryzacji, w którym użytkownik może się zalogować w zupełnie innej sieci i urządzenie które pyta może być w innej sieci. Proces autoryzacji opisaliśmy [tutaj](https://developer.allegro.pl/auth/#userDeviceFlow).

**Comment by PawelTaberski** [comment]
Adres przekierowania możesz zawsze zmienić, także nie jest to problem. Pisaliśmy o tym tutaj #915

**Comment by marcinja2** [comment]
OK ROZUMIEM JEST ŁATWA MOŻLIWOŚĆ ZMIANY. Ale w przypadku sieci prywatnej mam wpisać adresy ip prywatne?  CZY REDIRECT MA BYC ZAWSZE JEDNYM Z ADRESÓW PRZEKIEROWANIA podanym przy rejestracji aplikacji? ADRES URL TO ZA SZEROKIE POJĘCIE DLATEGO POTRZEBUJĘ PRZYKŁADU. Dla ludzi z dużym doświadczeniem sprawa może być oczywista ale nie każdy ma takowe :)

**Comment by PawelTaberski** [comment]
Nie wiem czy zapoznałeś się z naszą dokumentacją odnośnie tej autoryzacji, jest ona dostępna [tutaj](https://developer.allegro.pl/auth/#user).
Jeśli dobrze rozumiem Twoje pytanie to adres jaki podajesz przy rejestracji jest adresem przekierowania i musisz podać dokładnie ten sam przy autoryzacji użytkownika.

**Comment by juhacz** [comment]
Ile ważny jest identyfikator sesji korzystaj otrzymany dzięki doLoginWithAccessToken. Nigdzie nie znalazłem tej informacji.

**Comment by PawelTaberski** [comment]
@juhacz  Identyfikator sesji uzyskany zarówno przy pomocy doLogin, doLoginEnc, doLoginWithAccessToken jest ważny godzinę zgodnie z informacją w[ pomocy](https://allegro.pl/webapi/faq.php#faq_0).
